rt_bene_model_training/pytorch/train_model.py

```diff
@@ -0,0 +1,192 @@
+    _root_parser.add_argument('--no-benchmark', action='store_false', dest="benchmark")
+    _root_parser.add_argument('--num_io_workers', default=4, type=int)
+    _root_parser.add_argument('--k_fold_validation', action="store_true", dest="k_fold_validation")
+    _root_parser.add_argument('--all_dataset', action='store_false', dest="k_fold_validation")
+    _root_parser.add_argument('--accumulate_grad_batches', default=1, type=int)
+    _root_parser.add_argument('--seed', type=int, default=0)
+    _root_parser.set_defaults(benchmark=True)
+    _root_parser.set_defaults(augment=True)
+
+    _model_parser = TrainRTBENE.add_model_specific_args(_root_parser, root_dir)
+    _hyperparams = _model_parser.parse_args()
+
+    pl.seed_everything(_hyperparams.seed)
+
+    _train_subjects = []
+    _valid_subjects = []
+    if _hyperparams.dataset == "rt_bene":
+        if _hyperparams.k_fold_validation:
+            # 6 is discarded
+            _train_subjects.append([1, 2, 8, 10])
+            _train_subjects.append([3, 4, 7, 9])
+            _train_subjects.append([5, 12, 13, 14])
+
+            _valid_subjects.append([0, 11, 15, 16])
+            _valid_subjects.append([0, 11, 15, 16])
+            _valid_subjects.append([0, 11, 15, 16])
+        else:  # we want to train with the entire dataset
+            print('Training on the whole dataset - do not use the trained model for evaluation purposes!')
+            _train_subjects.append(
+                [0, 1, 2, 3, 4, 5, 8, 9, 10, 11, 12, 13, 14, 15, 16])  # 6 is discarded, 7 is used for validation
+            _valid_subjects.append([7])
+    else:
+        raise NotImplementedError("No other dataset is currently implemented")
+
+    for fold, (train_s, valid_s) in enumerate(zip(_train_subjects, _valid_subjects)):
+        # this is a hack to get class weights, i'm sure there's a better way fo doing it but I can't think of it
+        with h5py.File(_hyperparams.hdf5_file, mode="r") as _h5_f:
+            _class_weights = RTBENEH5Dataset.get_class_weights(h5_file=_h5_f, subject_list=train_s)
+
+        _model = TrainRTBENE(hparams=_hyperparams,
+                             train_subjects=train_s,
+                             validate_subjects=valid_s,
+                             class_weights=_class_weights)
+
+        checkpoint_callback = ModelCheckpoint(dirpath=_hyperparams.save_dir,
+                                              monitor='val_loss',
+                                              save_top_k=3,
+                                              filename=f'fold={fold}-' + '{epoch}-{val_loss:.2f}')
+
+        # start training
+        trainer = Trainer(gpus=_hyperparams.gpu,
+                          callbacks=[checkpoint_callback],
+                          precision=32,
+                          progress_bar_refresh_rate=1,
+                          min_epochs=5 if _hyperparams.augment else 3,
+                          max_epochs=20 if _hyperparams.augment else 5,
```

Comment:
Shouldn't there be fewer epochs in we augment?

Reply:
To be honest, the epochs length is rather arbitrary at the moment - this stems from if we dont augment in the Trainer but augment at data generation time to make them equal but I've been fiddling with the augment number and not augment number. 
I should make this an argument in the parser with some sensible defaults...